Documentation/Makefile

```diff
@@ -1,93 +1,45 @@
-
-check-requirements:
-	@set -e;								\
-	$(ECHO_CHECK) documentation dependencies...;				\
-	PYPKGS=$$(pip3 freeze);							\
-	for pkg in ${PIP_REQUIREMENTS}; do					\
-		echo $${PYPKGS} | grep -q $${pkg}				\
-		|| (echo "Documentation dependency '$${pkg}' not found.";	\
-		    echo "Run 'pip3 install --user -r Documentation/requirements.txt'";	\
-		    exit 2);							\
-	done
-
-cmdref:
-	$(QUIET) # We don't know what changed so recreate the directory
-	$(QUIET) -rm -rvf $(CMDREFDIR)/cilium*
-	@$(ECHO_GEN)cmdref/cilium
-	$(QUIET) ${CILIUMDIR}/cilium cmdref -d $(CMDREFDIR)
-	@$(ECHO_GEN)cmdref/cilium-bugtool
-	$(QUIET) ${BUGTOOLDIR}/cilium-bugtool cmdref -d $(CMDREFDIR)
-	@$(ECHO_GEN)cmdref/cilium-agent
-	$(QUIET) ${AGENTDIR}/cilium-agent --cmdref $(CMDREFDIR)
-	@$(ECHO_GEN)cmdref/cilium-health
-	$(QUIET) ${HEALTHDIR}/cilium-health --cmdref $(CMDREFDIR)
-	@$(ECHO_GEN)cmdref/cilium-operator
-	$(QUIET) ${OPERATORDIR}/cilium-operator --cmdref $(CMDREFDIR)
+clean:
+	-$(QUIET)rm -rf _build
 
-# Touch "$@.ok" if the docs build succeeds. Fail if there are errors
-# or output other than those including the following phrases:
-# "tabs assets" - These are always printed by the spelling check.
-# "misspelled words" - These will be explicitly checked below.
-# Finally, fail out if we can't remove the file that marks success.
-check-build: check-requirements
-	@$(ECHO_CHECK) documentation build...
-	@mkdir -p $(BUILDDIR)
-	$(QUIET)($(SPHINXBUILD) -b spelling -d "$(BUILDDIR)/doctrees" "$(SOURCEDIR)" "$(BUILDDIR)/spelling" -q -E $(SPHINXOPTS) 2>&1 && touch $@.ok) > $(BUILDDIR)/warnings.txt
-	@if grep -q -v $(IGNOREDWARNINGS) $(BUILDDIR)/warnings.txt; then \
-		echo "Please fix the following documentation warnings:"; \
-		grep -v $(IGNOREDWARNINGS) "$(BUILDDIR)/warnings.txt"; \
-		exit 1; \
-	fi
-	@if [ "$$(wc -l < "$(BUILDDIR)/spelling/output.txt")" -ne "0" ]; then \
-	        echo ""; \
-		echo "Please fix the following spelling mistakes:"; \
-		sed 's/^/* Documentation\//' $(BUILDDIR)/spelling/output.txt; \
-		echo "If the words are not misspelled, run:"; \
-		echo -n "Documentation/update-spelling_wordlist.sh "; \
-		echo "`sed -E 's/^([^:]+:){2} \((.*)\)/\2/g' $(BUILDDIR)/spelling/output.txt | sort -u | tr -d '\r\n' | sed "s/'/\\\\\\\\'/g"`"; \
-		echo ""; \
-		exit 1; \
-	fi
-	#                                                                                                               ^^^^^^ replace FOO'bar with FOO\'bar
-	@rm $@.ok 2>/dev/null
+builder-image:
+	$(QUIET)tar c requirements.txt Dockerfile \
+	  | docker build --tag cilium/docs-builder -
 
-# To use checklinks with `make render-docs`,
-# edit the line `%: Makefile check-requirements check-build checklinks`
-checklinks:
-	$(QUIET)$(SPHINXBUILD) -b linkcheck -d "$(BUILDDIR)/doctrees" "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS)
+DOCKER_RUN := docker container run --rm \
+		--workdir /src/Documentation \
+		--volume $(CURDIR)/..:/src \
+		--user "$(shell id -u):$(shell id -g)" \
```

Comment:
@joestringer this should work fine on Linux, would you mind to test please?

Reply:
Works on my setup (GNU Make 4.2.1).

Reply:
Works for me, thanks!